pkg/addons/default/aws_node.go

```diff
@@ -58,13 +58,18 @@ func UpdateAWSNode(rawClient kubernetes.RawClientInterface, region string) error
 			}
 		}
 
-		status, err := resource.CreateOrReplace()
+		status, err := resource.CreateOrReplace(plan)
 		if err != nil {
-			return err
+			return false, err
 		}
 		logger.Info(status)
 	}
 
+	if plan {
+		logger.Critical("(plan) %q is not up-to-date", AWSNode)
+		return true, nil
```

Comment:
What is this boolean that the function returns?

Reply:
it's to indicate pending changes, so that on the outside we can tell the user whether they want to run with `--approve` or it'd make no difference.